geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/MemberIdentifierImpl.java

```diff
@@ -235,23 +232,19 @@ public boolean hasUUID() {
   }
 
   public int compare(MemberIdentifier other) {
-    return this.compareTo(other, false, true);
+    return compareTo(other, false, true);
   }
 
   @Override
-  public int compareTo(MemberIdentifier other, boolean compareMemberData,
-      boolean compareViewIds) {
-    int myPort = getMembershipPort();
-    int otherPort = other.getMembershipPort();
-    if (myPort < otherPort) {
-      return -1;
-    }
-    if (myPort > otherPort) {
-      return 1;
+  public int compareTo(final @NotNull MemberIdentifier other, final boolean compareMemberData,
+      final boolean compareViewIds) {
+    int c = Integer.compare(getMembershipPort(), other.getMembershipPort());
```

Comment:
I see why you made `c` mutable—so that you could use it to capture the result of subsequent comparisons below.

If you wanted to make it `final` and make new `final` variables that would be ok too.

Reply:
Yeah, I toyed with that but it didn't really make it more readable in my opinion so I backed it out. I would just like to see us go to Java 9 or newer where most of these are replaced with built-ins.